src/ddmd/mtype.d

```diff
@@ -4558,103 +4558,23 @@ extern (C++) class TypeArray : TypeNext
         }
         if (ident == Id.reverse && (n.ty == Tchar || n.ty == Twchar))
         {
-            static __gshared const(char)** reverseName = ["_adReverseChar", "_adReverseWchar"];
-            static __gshared FuncDeclaration* reverseFd = [null, null];
-
-            deprecation(e.loc, "use std.algorithm.reverse instead of .reverse property");
-            int i = n.ty == Twchar;
-            if (!reverseFd[i])
-            {
-                auto params = new Parameters();
-                Type next = n.ty == Twchar ? Type.twchar : Type.tchar;
-                Type arrty = next.arrayOf();
-                params.push(new Parameter(0, arrty, null, null));
-                reverseFd[i] = FuncDeclaration.genCfunc(params, arrty, reverseName[i]);
-            }
-
-            Expression ec = new VarExp(Loc(), reverseFd[i], false);
-            e = e.castTo(sc, n.arrayOf()); // convert to dynamic array
-            auto arguments = new Expressions();
-            arguments.push(e);
-            e = new CallExp(e.loc, ec, arguments);
-            e.type = next.arrayOf();
+            error(e.loc, "use std.algorithm.reverse instead of .reverse property");
+            goto Lerror;
         }
         else if (ident == Id.sort && (n.ty == Tchar || n.ty == Twchar))
         {
-            static __gshared const(char)** sortName = ["_adSortChar", "_adSortWchar"];
-            static __gshared FuncDeclaration* sortFd = [null, null];
-
-            deprecation(e.loc, "use std.algorithm.sort instead of .sort property");
-            int i = n.ty == Twchar;
-            if (!sortFd[i])
-            {
-                auto params = new Parameters();
-                Type next = n.ty == Twchar ? Type.twchar : Type.tchar;
-                Type arrty = next.arrayOf();
-                params.push(new Parameter(0, arrty, null, null));
-                sortFd[i] = FuncDeclaration.genCfunc(params, arrty, sortName[i]);
-            }
-
-            Expression ec = new VarExp(Loc(), sortFd[i], false);
-            e = e.castTo(sc, n.arrayOf()); // convert to dynamic array
-            auto arguments = new Expressions();
-            arguments.push(e);
-            e = new CallExp(e.loc, ec, arguments);
-            e.type = next.arrayOf();
+            error(e.loc, "use std.algorithm.sort instead of .sort property");
+            goto Lerror;
```

Comment:
These first two branches can be removed, as the error is the same for both strings and arrays in this PR.